### PR TITLE
Fix for JAX unsigned ints

### PR DESCRIPTION
**Pull request by x46085**
## Uint64 Fix - StableHLO and IREE use **signless** integers

MLIR's integer type system distinguishes three kinds:
- **Signless** (`i64`) -- no sign semantics baked into the type; ops decide signedness
- **Signed** (`si64`) -- used in the StableHLO spec but rare in practice
- **Unsigned** (`ui64`) -- used in the StableHLO spec but rare in practice

In practice, the entire JAX → StableHLO → IREE pipeline overwhelmingly uses **signless `i64`**. IREE's own `StableHloToStdTypeConverter` includes `convertIntegerToSignless` which strips sign information during lowering. The IREE VM uses signless integers with per-operation sign semantics (e.g., `vm.cmp.gte.i32.u` for unsigned comparison on a signless register).

So at the hardware level, there is **no such thing** as a "uint64 tensor" in IREE. It's all `i64` -- the bit pattern is identical, and signedness is an operation-level concern.

## Why JAX makes this painful

JAX inherits NumPy's type promotion lattice, where `uint64` and `int64` are distinct types. The lattice has no `int128`, so the "join" of `uint64 + int64` goes **up to `float64`** -- a deliberate design choice documented in [JEP 9407](https://jax.readthedocs.io/en/latest/jep/9407-type-promotion.html).

This means any operation that mixes `uint64` with `int64` (scatter indices, `jnp.where`, arithmetic) produces `float64` intermediates, which then fail index verification. This is a **JAX design constraint**, not an Elodin bug.

## What the fix actually does

The fix aligns Elodin's internal type representations with the target backend's type system:

```
Customer declares:  ComponentType.U64  (semantic intent: "this is an unsigned counter")
PrimTypeExt maps:   U64 → S64          (JAX-compatible: "this is a 64-bit integer")
StableHLO emits:    i64                (signless, as IREE expects)
IREE compiles:      i64                (signless, hardware-native)
Bit pattern:        identical at every stage
```

The customer's **values are preserved exactly**. A `u64` value of `42` has the same bit representation as an `i64` value of `42`. The entire valid range of entity IDs, state codes, and counters (which are always small positive numbers) maps identically between signed and unsigned 64-bit integers.

## What would "true unsigned" require?

To support `uint64` as a distinct type through the full pipeline, you would need:

1. **JAX to not promote `uint64 + int64 → float64`** -- this is fundamental to JAX's type system and cannot be overridden. There's no config, no hook, no monkey-patch that changes the lattice.

2. **Every scatter, gather, and index operation to accept `uint64` indices** -- JAX explicitly rejects non-integer-typed indices, and `float64` (the result of promotion) is rejected.

3. **StableHLO to preserve `ui64` through lowering** -- possible in theory, but IREE's conversion passes strip it to signless `i64` anyway.

So even if you threaded `uint64` through JAX somehow, IREE would convert it back to signless `i64` at compilation time. The bit pattern would be identical to what we produce now.

## The `PrimTypeExt` pattern is standard practice

This is the same pattern used by:
- **IREE itself**: `RemoveSignTypeConverter` strips signedness from StableHLO types
- **XLA/TF**: internally uses signless integers in HLO
- **MLIR Linalg**: the `LinalgTypeConverter` extends `RemoveSignTypeConverter`

The Elodin approach just applies this normalization one step earlier (at the Python/JAX boundary) rather than waiting for the MLIR pass to do it. This avoids the JAX type promotion landmine entirely.

## What the customer should know

The customer can continue declaring `ComponentType.U64` -- that's the schema-level type that describes the semantic intent ("this is unsigned data"). Under the hood, Elodin transparently presents it as `int64` to JAX, because that's what the entire MLIR/IREE backend actually uses. Their values are bit-for-bit preserved. There is no loss of precision, range, or correctness for any value that fits in both `u64` and `i64` (which is everything up to `2^63 - 1` -- far beyond any entity ID or state code).

### CODE REVIEW

File: libs/nox-py/src/world_builder.rs [hunk 1]
```diff
@@ -1812,31 +1812,31 @@ impl WorldBuilder {
 
             match schema.prim_type {
                 PrimType::U8 => {
-                    let slice = <[u8]>::ref_from_bytes(data).unwrap();
+                    let slice = <[i8]>::ref_from_bytes(data).unwrap();
```
[the remark this answers is not in the document]
Comment (anderspitman) [reply libs/nox-py/src/world_builder.rs]: I'm not too worried about U64, but this seems like a valid concern for the smaller types, especially U8.

[the remark this answers is not in the document]
Comment (x46085) [reply libs/nox-py/src/world_builder.rs]: The PrimTypeExt pattern is standard practice
This is the same pattern used by:

IREE itself: RemoveSignTypeConverter strips signedness from StableHLO types
XLA/TF: internally uses signless integers in HLO
MLIR Linalg: the LinalgTypeConverter extends RemoveSignTypeConverter
The Elodin approach just applies this normalization one step earlier (at the Python/JAX boundary) rather than waiting for the MLIR pass to do it. This avoids the JAX type promotion landmine entirely.

**Review by anderspitman — APPROVED** [review]
